### DISCUSSION TITLE
Keep previous page mounted after changing route

### DISCUSSION THREAD

**Discussion by ArmanEskandari** [vercel/next.js#46612] [Help]
Is there a way to prevent the next js router from unmounting the previous page?

**Comment by icyJoseph** [comment]
Not built-in, and before going down that path I'd recommend defining exactly your requirements.

> If you use dynamic routes, you can, sort of, cheat your way around

It's not impossible though.

Frameworks like `framer-motion`, internally implement something called `AnimatePresence`, which basically keeps an unmounted child. 

- https://github.com/framer/motion/blob/main/packages/framer-motion/src/components/AnimatePresence/index.tsx

So it is technically possible. I'd recommend setting up a bunch of unit tests, as close as possible to what you want to achieve and go from there.

Just yesterday I helped someone who was facing an issue where the component started to consume router information, that did not belong to it.

- https://github.com/vercel/next.js/discussions/46534#discussioncomment-5158105

Several years ago I wrote an article about playing with the `hidden` HTML attribute (fancy talk for display none), to keep content mounted, but invisible.

There's also talk about having this built into React, so that some content can be rendered off-screen, https://reactjs.org/blog/2022/03/29/react-v18.html#what-is-concurrent-react:

> ...React should be able to restore the previous screen in the same state it was in before. In an upcoming minor, we’re planning to add a new component called <Offscreen> that implements this pattern. Similarly, you’ll be able to use Offscreen to prepare new UI in the background so that it’s ready before the user reveals it...
